editoast/src/views/train_schedule/mod.rs

```diff
@@ -335,9 +335,32 @@ async fn get_results(
         .ok_or(TrainScheduleError::TimetableNotFound {
             timetable_id: query.timetable_id,
         })?;
-    let schedules = timetable.get_train_schedules(db_pool.clone()).await?;
-
-    if schedules.is_empty() {
+    let infra_version = timetable
+        .infra_version_from_timetable(db_pool.clone())
+        .await;
+    let pool = db_pool.clone();
+    let mut schedules = timetable.get_train_schedules(pool.clone()).await?;
+    let schedules = block::<_, Result<_>>(move || {
```

Comment:
Can you try to factorize this part (and the STDCM one) to keep that function smaller?